platform/ui/src/contextProviders/ViewportGridProvider.tsx

```diff
@@ -25,20 +39,59 @@ const DEFAULT_STATE = {
       viewportLabel: null,
     },
   ],
-  activeViewportIndex: 0,
-  cachedLayout: {},
 };
 
 export const ViewportGridContext = createContext(DEFAULT_STATE);
 
+const findOrCreate = (index, positionId, cached) => ({
+  ...(cached.byPosition?.[positionId] || {}),
+});
+
 /**
- * Given the flatten index, and rows and column, it returns the
- * row and column index
+ * Find a viewport to re-use, and then set the viewportId
+ *
+ * @param idSet
+ * @param viewport
+ * @param stateViewports
+ * @returns
  */
-const unravelIndex = (index, numRows, numCols) => {
-  const row = Math.floor(index / numCols);
-  const col = index % numCols;
-  return { row, col };
+const reuseViewport = (idSet, viewport, stateViewports) => {
```

Comment:
I don't understand this function,
- what is `idSet` ? naming is not good
- then it loops through stateViewports and set some to true
- then it loops 10000 times? 

what is going on?